lib/screens/UsersProfileScreen.dart

```diff
@@ -0,0 +1,1576 @@
+                      color: bottomRowIcon,
+                    ),
+                    const Text(
+                      'Shopping Cart',
+                      style: TextStyle(fontSize: 12, color: bottomRowIcon),
+                      textAlign: TextAlign.center,
+                    )
+                  ],
+                ),
+              ),
+              SizedBox(
+                width: MediaQuery.of(context).size.width / 4,
+                child: Column(
+                  crossAxisAlignment: CrossAxisAlignment.center,
+                  children: <Widget>[
+                    IconButton(
+                      onPressed: () {},
+                      icon: Icon(Icons.person),
+                      iconSize: 55,
+                      color: mainScheme,
+                    ),
+                    const Text(
+                      'User Profile',
+                      style: TextStyle(fontSize: 12, color: mainScheme),
+                      textAlign: TextAlign.center,
+                    )
+                  ],
+                ),
+              ),
+            ],
+          ),
+        ),
+      ),
+    );
+  }
+
+  bool areFieldsValid() {
+    bool toReturn = true;
+    if (_firstName.value.text.isEmpty) {
+      toReturn = false;
+      unfilledFirstName = true;
+    }
+    if (_lastName.value.text.isEmpty) {
+      toReturn = false;
+      unfilledLastName = true;
+    }
+    if (_email.value.text.isEmpty) {
+      toReturn = false;
+      unfilledEmail = true;
+    }
+    if (_password.value.text.isEmpty) {
+      toReturn = false;
+      unfilledPassword = true;
+    }
+    if (_confirmPassword.value.text.isEmpty) {
+      toReturn = false;
+      unfilledConfirmPassword = true;
+    }
+    if (!isEmail(_email.value.text)) {
+      toReturn = false;
+      errorMessage = 'Email must be in valid form';
+      unfilledEmail = true;
+    }
+    return toReturn;
+  }
+
+  bool passwordsMatch() {
+    if (_password.value.text != user.password) {
+      errorMessage = 'Passwords do not match';
+      return false;
+    }
+    return true;
+  }
+
+  void clearFields() {
+    unfilledEmail = false;
+    unfilledPassword = false;
+    _email.clear();
+    _password.clear();
+  }
+
+  Future<bool> status(http.Response res) async {
+    bool accepted = true;
+    while (accepted) {
+      switch (res.statusCode) {
+        case 200:
+          return true;
+        case 400:
+          print("Incorrect formatting!");
+          return false;
+        case 401:
+          errorMessage = 'Access token missing';
+          return false;
+        case 403:
+          final changeToken = await Authentication.refreshToken();
+          if (refreshTokenStatus(changeToken)) {
+            var tokens = json.decode(changeToken.body);
+            user.accessToken = tokens['accessToken'];
+            user.refreshToken = tokens['refreshToken'];
+          }
+          else {
+            print('something went wrong!');
```

Comment:
If you end up here, you should try logging the user in again using the saved info and if that fails, redirect to startup